### PR TITLE
Add support for List negative index

### PR DESCRIPTION
**Pull request by thanhnguyen-aws**
*Description of changes:* This PR add support for List negative index.

How: When `n < 0` and `n >= - List_len(someList)`, `someList[n]` will be converted into `someList[n + List_len(someList)]`

Limitation: This PR does not support negative index in List Slice

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: Strata/Languages/Python/PythonRuntimeLaurelPart.lean [hunk 1]
```diff
@@ -439,15 +439,18 @@ function DictStrAny_insert (d : DictStrAny, key: string, val: Any) : DictStrAny
...
                   || (OptionInt..isOptNone(Any..stop!(index)))))
 {
   if Any..isfrom_DictStrAny(dictOrList) then
     DictStrAny_get(Any..as_Dict!(dictOrList), Any..as_string!(index))
   else if Any..isfrom_ListAny(dictOrList) && Any..isfrom_int(index) then
-    List_get(Any..as_ListAny!(dictOrList), Any..as_int!(index))
+    if Any..as_int!(index) >= 0 then
+      List_get(Any..as_ListAny!(dictOrList), Any..as_int!(index))
+    else
+      List_get(Any..as_ListAny!(dictOrList), List_len(Any..as_ListAny!(dictOrList)) + Any..as_int!(index))
```
Comment (tautschnig) [inline Strata/Languages/Python/PythonRuntimeLaurelPart.lean]: The negative-index conversion logic is duplicated between `Any_get` and `Any_get!`. Consider extracting a helper function to avoid the duplication:

```
function List_get_with_neg_index (l: ListAny, index: int): Any
  requires index >= - List_len(l) && index < List_len(l)
{
  if index >= 0 then List_get(l, index)
  else List_get(l, List_len(l) + index)
};
```

Then both `Any_get` and `Any_get!` can call `List_get_with_neg_index(Any..as_ListAny!(dictOrList), Any..as_int!(index))`.

File: Strata/Languages/Python/PythonRuntimeLaurelPart.lean [hunk 2]
```diff
@@ -439,15 +439,18 @@ function DictStrAny_insert (d : DictStrAny, key: string, val: Any) : DictStrAny
 
 function Any_get (dictOrList: Any, index: Any): Any
   requires  (Any..isfrom_DictStrAny(dictOrList) && Any..isfrom_str(index) && DictStrAny_contains(Any..as_Dict!(dictOrList), Any..as_string!(index))) ||
-            (Any..isfrom_ListAny(dictOrList) && Any..isfrom_int(index) && Any..as_int!(index) >= 0 && Any..as_int!(index) < List_len(Any..as_ListAny!(dictOrList)))||
+            (Any..isfrom_ListAny(dictOrList) && Any..isfrom_int(index) && Any..as_int!(index) >= - List_len(Any..as_ListAny!(dictOrList)) && Any..as_int!(index) < List_len(Any..as_ListAny!(dictOrList)))||
```
Comment (tautschnig) [inline Strata/Languages/Python/PythonRuntimeLaurelPart.lean:456]: The precondition change is correct: `index >= -List_len(list)` allows indices from `-len` to `len-1`, matching Python semantics.

However, `Any_set` (line ~477) and `Any_set!` (line ~486) still require `index >= 0`. In Python, `l[-1] = 42` is valid. This inconsistency means negative-index writes will fail verification.

File: StrataTest/Languages/Python/tests/test_list.py [hunk 3]
```diff
@@ -6,6 +6,8 @@
 
 assert n == 30
 
+assert numbers[-2] == 40
```
Comment (tautschnig) [inline StrataTest/Languages/Python/tests/test_list.py:9]: Consider also adding:
- Edge case: `assert numbers[-5] == 10` (first element via negative index, i.e., `-len`)
- A test for negative-index write if `Any_set` is updated: `numbers[-1] = 99; assert numbers[-1] == 99`
